### PR TITLE
Update tests to use "llvm-cpu" and "local-task" names consistently.

### PR DESCRIPTION
**Pull request by ScottTodd**
Progress on https://github.com/iree-org/iree/issues/9930

Also fixed a few test suite names to match `${COMPILER_TARGET}_${HAL_DEVICE}` naming (these had drifted a bit).

### CODE REVIEW

File: docs/developers/developing_iree/testing_guide.md [hunk 1]
```diff
@@ -219,13 +219,13 @@ drivers).
 With CMake, run this from the build directory:
 
 ```shell
-$ ctest -R tests/e2e/xla_ops/check_vmvx_vmvx_floor.mlir
+$ ctest -R tests/e2e/xla_ops/check_vmvx_local-task_floor.mlir
```
Comment (benvanik) [inline docs/developers/developing_iree/testing_guide.md:222]: it's annoying to have mixed -/_ in these - can we just make them all _?

File: runtime/bindings/python/iree/runtime/system_api.py [hunk 2]
```diff
@@ -47,7 +47,7 @@
 
 # Mapping from IREE target backends to their corresponding drivers.
 TARGET_BACKEND_TO_DRIVER = {
-    "dylib-llvm-aot": "local-task",
+    "llvm-cpu": "local-task",
```
Comment (ScottTodd) [inline runtime/bindings/python/iree/runtime/system_api.py:50]: I guess this isn't load bearing (any more)? Seems like I can make this default mapping change independent of other changes.

File: docs/developers/developing_iree/testing_guide.md (hunk 1 above)
Comment (ScottTodd) [reply docs/developers/developing_iree/testing_guide.md:222]: `check_llvm-cpu_local-task` follows `check_${COMPILER_TARGET}_${HAL_DEVICE}`, with `llvm-cpu` and `local-task` names matching what the compiler and runtime expect exactly. We could diverge from the code for consistency in build system target naming, yeah.

Comment (GMNGeoffrey) [reply docs/developers/developing_iree/testing_guide.md:222]: I'd rather we have consistency with the driver naming, personally. Then a search for "llvm-cpu" turns up all usages and we don't have to remember it's sometimes "llvm_cpu". Additionally, this gives us delimiters such that "local-task" is clearly all one thing and it's not "vmvx_local".

Comment (ScottTodd) [reply docs/developers/developing_iree/testing_guide.md:222]: @benvanik how do you feel about that? I like being able to find-replace too, even though target name style is important too...

**Comment by ScottTodd** [comment]
Ping?

* One unresolved thread about mixed `_` and `-`, with opinions on both sides
* pytype check failure can be ignored here

**Pull request merged**